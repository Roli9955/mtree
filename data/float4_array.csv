id;val
0;-0.999999
1;-0.999999,0
2;-0.999999,0,0.999999