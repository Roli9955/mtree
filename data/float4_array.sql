/* contrib/mtree_gist/data/float4_array.sql */

DROP TABLE IF EXISTS FLOAT4_ARRAY_TEST CASCADE;
CREATE TABLE FLOAT4_ARRAY_TEST (
  id  INTEGER,
  val mtree_float_array
);
COPY FLOAT4_ARRAY_TEST(id, val) FROM '/home/postgres/test_files/float4_array.csv' DELIMITER ';' CSV HEADER;

DROP INDEX IF EXISTS FLOAT4_ARRAY_TEST_IDX CASCADE;
CREATE INDEX FLOAT4_ARRAY_TEST_IDX ON FLOAT4_ARRAY_TEST USING GiST (val mtree_float_array_opclass);

SET enable_seqscan TO OFF;

SELECT COUNT(*) FROM FLOAT4_ARRAY_TEST;

SELECT * FROM FLOAT4_ARRAY_TEST;

SELECT id, val, (val <-> '0') AS dst FROM FLOAT4_ARRAY_TEST ORDER BY (val <-> '0'), id;

SELECT * FROM FLOAT4_ARRAY_TEST WHERE val #<# '0';
