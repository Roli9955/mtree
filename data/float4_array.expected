DROP TABLE
CREATE TABLE
COPY 3
DROP INDEX
CREATE INDEX
SET
 id |             val             
----+-----------------------------
  0 | -0.999999
  1 | -0.999999,0.000000
  2 | -0.999999,0.000000,0.999999
(3 rows)

 id |             val             |   dst    
----+-----------------------------+----------
  2 | -0.999999,0.000000,0.999999 |        0
  0 | -0.999999                   | 0.999999
  1 | -0.999999,0.000000          | 0.999999
(3 rows)

 id |             val             
----+-----------------------------
  2 | -0.999999,0.000000,0.999999
(1 row)

